src/messages/tests.rs

```diff
@@ -0,0 +1,410 @@
+use rand::thread_rng;
+
+use super::*;
+
+use crate::messages::validate::{MsgErr, Validate};
+
+use crate::frost;
+
+use serde_json;
+use std::convert::TryFrom;
+
+#[test]
+fn validate_version() {
+    // A version number that we expect to be always invalid
+    const INVALID_VERSION: u8 = u8::MAX;
+
+    let signer1 = ParticipantId::Signer(0);
+    let dealer = ParticipantId::Dealer;
+
+    let validate = Validate::validate(&Header {
+        version: MsgVersion(INVALID_VERSION),
+        sender: dealer.clone(),
+        receiver: signer1.clone(),
+    })
+    .err()
+    .expect("an error");
+
+    assert_eq!(validate, MsgErr::WrongVersion);
+
+    let validate = Validate::validate(&Header {
+        version: constants::BASIC_FROST_SERIALIZATION,
+        sender: dealer,
+        receiver: signer1,
+    })
+    .err();
+
+    assert_eq!(validate, None);
```

Comment:
Similarly, this can become:
```suggestion
    let msg = Header {
        version: constants::BASIC_FROST_SERIALIZATION,
        sender: dealer,
        receiver: signer1,
    };
    assert_eq!(Validate::validate(&msg), Ok(&msg));
```

Again, this is just a suggestion for future tests.